RELEASE_NOTES.md

```diff
@@ -0,0 +1,10 @@
+Release Notes
+=============
+
+v1.0.0-alpha.3
```

Comment:
[consider] I think the version should be v1.0.0-alpha.X or vX.X.X as per [our CHANGES process](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/contributing/CHANGES.md).

Also I _think_ the commit component should be `[Docs]`.  From [our COMMITS process](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/contributing/COMMITS.md#appendix---code-components)

> Release For commits bumping versions in preparation for a release.

vs. 

> Docs For work in the (distribution) Doxygen documentation or in-repository Markdown files.

They're both kinda relevant. I would swing toward `[Docs]`, since presumably this isn't "in preparation for a release" (since it's already happened).

Reply:
> I think the version should be v1.0.0-alpha.X or vX.X.X as per [our CHANGES process](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/contributing/CHANGES.md).

This is true if we were making a change for a future release, but we've just done the alpha 3 release, we're more documenting it after the fact, wouldn't the next change introduce a .X version?

Agree on docs, will change.

Reply:
> we're more documenting it after the fact

Good point, absolutely right, ignore me.

Reply:
Changed to docs tag in rebase.